iib/common/tracing.py

```diff
@@ -81,37 +81,37 @@ def __new__(cls):
 
 
 def instrument_tracing(
-    func=None,
     span_name: str = '',
     attributes: Dict = {},
 ):
     """
     Instrument tracing for a function.
 
-    :param func: The function to be decorated.
     :param span_name: The name of the span to be created.
     :param attributes: The attributes to be added to the span.
     :return: The decorated function or class.
     """
-    log.info('Instrumenting span for %s', span_name)
-    tracer = trace.get_tracer(__name__)
-    context = None
-    if trace.get_current_span():
-        context = trace.get_current_span().get_span_context()
-    else:
-        context = propagator.extract(carrier={})
 
     def decorator_instrument_tracing(func):
         @functools.wraps(func)
         def wrapper(*args, **kwargs):
             if not os.getenv('IIB_OTEL_TRACING', '').lower() == 'true':
                 return func(*args, **kwargs)
-            log.debug('Context inside %s: %s', span_name, context)
+
+            log.info('Instrumenting span for %s', span_name)
+            tracer = trace.get_tracer(__name__)
+            log.info('Tracer: %s', tracer)
+            if trace.get_current_span():
+                context = trace.get_current_span().get_span_context()
+            else:
+                context = propagator.extract(carrier={})
+
+            log.info('Context inside %s: %s', span_name, context)
             if kwargs.get('traceparent'):
                 log.debug('traceparent is %s' % str(kwargs.get('traceparent')))
                 carrier = {'traceparent': kwargs.get('traceparent')}
                 trace_context = propagator.extract(carrier)
-                log.debug('Context is %s', trace_context)
+                log.info('Context is %s', trace_context)
```

Comment:
Hey @lipoja , should these log message be visible in build logs?

Reply:
I am sorry - what build log are you referring to?

Reply:
I was debugging it and `kwargs` is always empty :( ... so we do not hit this part of code

Reply:
Thanks, Jan, I checked more and got this bit.
Should this be fixed as well? and will it be part of this PR?

Reply:
it will not be part of this PR - I think it needs more investigation / understanding.